src/Game/Client.java

```diff
@@ -1314,6 +1319,10 @@ public static void disconnect_hook() {
     player_id = -1;
     knowWhoIAm = false;
     Client.tipOfDay = -1;
+    if (Settings.LOG_LAG.get(Settings.currentProfile)) {
+      Logger.Lag("Disconnect", Replay.timestamp);
+      Logger.finalizeLagLog();
+    }
```

Comment:
If someone has this setting turned on before logging in then disables the setting before logging out, it won't end up calling `finalizeLagLog()`, which would keep the PrintWriter open.

One way to deal with this is by moving the `finalizeLagLog()` invocation outside of the conditional here:

```java
if (Settings.LOG_LAG.get(Settings.currentProfile)) {
  Logger.Lag("Disconnect", Replay.timestamp);
}
Logger.finalizeLagLog();
```

and adding a null check to the finalize method:

```java
public static void finalizeLagLog() {
  if (m_lagWriter == null) {
    return;
  }

  try {
    m_lagWriter.close();
  } catch (Exception e) {
    Error("Could not stop logging lag.");
  }
  Info("Finished lag logging @ " + m_lagWriter_filename);
}
```

Doing so would guarantee that the PrintWriter gets closed no matter what.